src/components/learner-credit-management/invite-modal/InviteMembersModalWrapper.jsx

```diff
@@ -0,0 +1,121 @@
+
+import { useBudgetId, useSubsidyAccessPolicy } from '../data';
+import InviteModalContent from './InviteModalContent';
+import SystemErrorAlertModal from '../cards/assignment-allocation-status-modals/SystemErrorAlertModal';
+import LmsApiService from '../../../data/services/LmsApiService';
+import { BudgetDetailPageContext } from '../BudgetDetailPageWrapper';
+
+const InviteMembersModalWrapper = ({ isOpen, close }) => {
+  const { subsidyAccessPolicyId } = useBudgetId();
+  const { data: subsidyAccessPolicy } = useSubsidyAccessPolicy(subsidyAccessPolicyId);
+  const [learnerEmails, setLearnerEmails] = useState([]);
+  const [canInviteMembers, setCanInviteMembers] = useState(false);
+  const [inviteButtonState, setInviteButtonState] = useState('default');
+  const [isSystemError, openSystemErrorModal, closeSystemErrorModal] = useToggle(false);
+  const {
+    successfulInvitationToast: { displayToastForInvitation },
+  } = useContext(BudgetDetailPageContext);
+
+  const handleCloseInviteModal = () => {
+    close();
+    setInviteButtonState('default');
+  };
+
+  // Callback function for when emails are changed in the
+  // child AssignmentModalContent component. Must be memoized as
+  // the function is used within a `useEffect`'s dependency array.
+  const handleEmailAddressesChanged = useCallback((
+    value,
+    { canInvite = false } = {},
+  ) => {
+    setLearnerEmails(value);
+    setCanInviteMembers(canInvite);
+  }, []);
+
+  const handleInviteMembers = async () => {
+    setInviteButtonState('pending');
+    const payload = snakeCaseObject({
+      learnerEmails,
+    });
+
+    try {
+      const groupUuid = subsidyAccessPolicy.groupAssociations[0];
+      const response = await LmsApiService.inviteEnterpriseLearnersToGroup(groupUuid, payload);
+      if (response.status === 201) {
+        const totalLearnersInvited = response.data.records_processed;
+        setInviteButtonState('complete');
+        handleCloseInviteModal();
+        displayToastForInvitation({
+          totalLearnersInvited,
+        });
+      }
+    } catch (err) {
+      setInviteButtonState('error');
+      openSystemErrorModal();
+    }
+  };
+
+  return (
+    <>
+      <FullscreenModal
+        className="stepper-modal bg-light-200"
+        title="New members"
+        isOpen={isOpen}
+        onClose={() => {
+          handleCloseInviteModal();
+        }}
+        footerNode={(
+          <ActionRow>
+            <Button
+              variant="tertiary"
+              as={Hyperlink}
+              destination={getConfig().ENTERPRISE_SUPPORT_URL}
+              showLaunchIcon
+              target="_blank"
+            >
+              Help Center: Invite Budget Members
+            </Button>
+            <ActionRow.Spacer />
+            <Button variant="tertiary" onClick={handleCloseInviteModal}>Cancel</Button>
+            <StatefulButton
+              labels={{
+                default: 'Invite',
```

Comment:
I think personally it would make it less readable, and I wanted to follow the pattern set from this https://github.com/openedx/frontend-app-admin-portal/blob/135cb6ec381b0a36db7b19d214030aaa4dfba4b3/src/components/learner-credit-management/cards/NewAssignmentModalButton.jsx#L241

Reply:
Also, worth nothing that if/when i18n is introduced into this MFE, any English strings stored in constants will need to be replaced with `intl.formatMessage` or `FormattedMessage`, etc.